### PR TITLE
OCM-7594 | test: Fix proxy realated function

### PR DESCRIPTION
**Pull request by yingzhanredhat**
```
[yingzhan@localhost ocm-common]$ go run main/main.go 
INFO[2024-04-25 14:58:54] Trying to list subnets of the vpc            
INFO[2024-04-25 14:58:55] Got 2 subnets                                
INFO[2024-04-25 14:58:55] subnet-02bec8d3a5123a2fd      10.0.128.0/20   us-west-2a       
INFO[2024-04-25 14:58:55] subnet-071e6a8e88668dc6a      10.0.0.0/20     us-west-2a       
INFO[2024-04-25 14:58:55] Got custom rt rtb-0236ac8e82155d80a          
INFO[2024-04-25 14:58:55] Got main association for rt rtb-0fb92f3f0b7239360 
INFO[2024-04-25 14:58:55] Got custom rt rtb-054bf10378e6713a8          
INFO[2024-04-25 14:58:56] Create security group sg-0824a2ed5f41efc42 success for vpc-0d156715c2f8f530b 
INFO[2024-04-25 14:58:57] Tag resource sg-0824a2ed5f41efc42 successfully 
INFO[2024-04-25 14:58:57] Created tagged security group with ID sg-0824a2ed5f41efc42 
INFO[2024-04-25 14:58:58] Authorize security group success sg-0824a2ed5f41efc42 
INFO[2024-04-25 14:58:58] Authorize security group success sg-0824a2ed5f41efc42 
INFO[2024-04-25 14:58:58] Authorize SG sg-0824a2ed5f41efc42 prepared successfully for proxy. 
INFO[2024-04-25 14:58:59] Create key pair success: key-0307731ce9a520a84 
INFO[2024-04-25 14:58:59] create key pair: key-0307731ce9a520a84 successfully 
INFO[2024-04-25 14:58:59] Tag resource key-0307731ce9a520a84 successfully 
INFO[2024-04-25 14:59:01] Waiting for below instances ready: i-008cd837ad061fe31 
INFO[2024-04-25 14:59:01] Check instances status of i-008cd837ad061fe31 
INFO[2024-04-25 14:59:01] Instance ID i-008cd837ad061fe31 is in status of not-applicable 
INFO[2024-04-25 14:59:01] Instance ID i-008cd837ad061fe31 is in state of pending 
INFO[2024-04-25 15:00:01] Check instances status of i-008cd837ad061fe31 
INFO[2024-04-25 15:00:02] Instance ID i-008cd837ad061fe31 is in status of initializing 
INFO[2024-04-25 15:00:02] Instance ID i-008cd837ad061fe31 is in state of running 
INFO[2024-04-25 15:00:02] All instances running                        
INFO[2024-04-25 15:00:02] Launch proxy instance i-008cd837ad061fe31 succeed 
INFO[2024-04-25 15:00:03] Tag resource i-008cd837ad061fe31 successfully 
INFO[2024-04-25 15:00:03] Allocated EIP eipalloc-069f12cb2bbb90915 with ip 34.214.97.157 
INFO[2024-04-25 15:00:03] Successfully allocated EIP: 34.214.97.157    
INFO[2024-04-25 15:00:05] Successfully allocated 34.214.97.157 with instance i-008cd837ad061fe31.
        allocation id: eipalloc-069f12cb2bbb90915, association id: eipassoc-01ad1e5907316965e 
INFO[2024-04-25 15:00:05] Prepare EIP successfully for the proxy preparation. Launch with IP: 34.214.97.157 
INFO[2024-04-25 15:02:11] HTTP PROXY: http://10.0.9.63:8080            
INFO[2024-04-25 15:02:11] HTTPs PROXY: https://10.0.9.63:8080          
INFO[2024-04-25 15:02:11] CA FILE PATH: /home/yingzhan/autotest/ocm-common/main/mitm-ca.pem 
INFO[2024-04-25 15:02:14] Terminate instances i-008cd837ad061fe31 successfully 
INFO[2024-04-25 15:02:14] Waiting for below instances terminated: i-008cd837ad061fe31  
INFO[2024-04-25 15:02:14] Check instances status of i-008cd837ad061fe31 
INFO[2024-04-25 15:02:14] Instance ID i-008cd837ad061fe31 is in status of not-applicable 
INFO[2024-04-25 15:02:14] Instance ID i-008cd837ad061fe31 is in state of shutting-down 
INFO[2024-04-25 15:03:14] Check instances status of i-008cd837ad061fe31 
INFO[2024-04-25 15:03:15] Instance ID i-008cd837ad061fe31 is in status of not-applicable 
INFO[2024-04-25 15:03:15] Instance ID i-008cd837ad061fe31 is in state of shutting-down 
INFO[2024-04-25 15:04:15] Check instances status of i-008cd837ad061fe31 
INFO[2024-04-25 15:04:15] Instance ID i-008cd837ad061fe31 is in status of not-applicable 
INFO[2024-04-25 15:04:15] Instance ID i-008cd837ad061fe31 is in state of terminated 
INFO[2024-04-25 15:04:15] Terminating instances i-008cd837ad061fe31 successfully 
INFO[2024-04-25 15:04:16] Delete key pair success                      
INFO[2024-04-25 15:04:16] delete key pair successfully                 
```

### CODE REVIEW

**Comment by yingzhanredhat** [comment]
@xueli181114 Would you help review?

File: pkg/test/vpc_client/proxy.go [hunk 1]
```diff
@@ -46,20 +47,39 @@ func (vpc *VPC) LaunchProxyInstance(imageID string, zone string, sshKey string)
...
 	}
 	tags := map[string]string{
 		"Name": CON.ProxyName,
 	}
+	_, err = vpc.AWSClient.TagResource(*key.KeyPairId, tags)
+	if err != nil {
+		log.LogError("Add tag for key pair %s failed %s", *key.KeyPairId, err)
+		return inst, "", "", err
+	}
+	sshKey, err := file.WriteToFile(*key.KeyMaterial, "proxyKeyPair.pem", privateKeyPath)
```
Comment (xueli181114) [inline pkg/test/vpc_client/proxy.go:64]: Let's not hardcode the keyName

Comment (yingzhanredhat) [reply pkg/test/vpc_client/proxy.go]: Fixed it. Would you help review it again?

**Review by ciaranRoche — APPROVED** [review]
/lgtm

**Pull request merged**